From ff0c18190e95c331f753802adb0f1592ab6998d4 Mon Sep 17 00:00:00 2001
From: Denys Vlasenko <dvlasenk@redhat.com>
Date: Wed, 27 Mar 2013 13:03:00 +0100
Subject: [RHEL6 LIBREPORT PATCH 80/94] reporter-rhtsupport: skip hints check
 if uploaded data is really large

Hints check uploads the entire problem directory tarball.
If the tarball is hundreds of megabytes, that is not a good idea.

Perhaps "hints" thing on the server one day will be changed so that
it accepts a truncated set of data.

Signed-off-by: Denys Vlasenko <dvlasenk@redhat.com>
Signed-off-by: Jiri Moskovcak <jmoskovc@redhat.com>
---
 src/plugins/reporter-rhtsupport.c | 81 +++++++++++++++++++++------------------
 1 file changed, 43 insertions(+), 38 deletions(-)

diff --git a/src/plugins/reporter-rhtsupport.c b/src/plugins/reporter-rhtsupport.c
index af4722b..94b0634 100644
--- a/src/plugins/reporter-rhtsupport.c
+++ b/src/plugins/reporter-rhtsupport.c
@@ -24,6 +24,8 @@
 #include "abrt_rh_support.h"
 #include "reporter-rhtsupport.h"
 
+#define QUERY_HINTS_IF_SMALLER_THAN  (8*1024*1024)
+
 static report_result_t *get_reported_to(const char *dump_dir_name)
 {
     struct dump_dir *dd = dd_opendir(dump_dir_name, /*flags:*/ 0);
@@ -376,50 +378,53 @@ int main(int argc, char **argv)
         goto ret;
     }
 
-//TODO: fetch size, use it to decide whether to fetch hints
+    off_t tempfile_size = stat_st_size_or_die(tempfile);
 
-    /* Check for hints and show them if we have something */
-    result = get_rhts_hints(url,
-            login,
-            password,
-            ssl_verify,
-            tempfile
-    );
+    if (tempfile_size <= QUERY_HINTS_IF_SMALLER_THAN)
+    {
+        /* Check for hints and show them if we have something */
+        result = get_rhts_hints(url,
+                login,
+                password,
+                ssl_verify,
+                tempfile
+        );
 #if 0 /* testing */
-    log("ERR:%d", result->error);
-    log("MSG:'%s'", result->msg);
-    log("BODY:'%s'", result->body);
-    result->error = 0;
-    result->body = xstrdup(
-    "<?xml version=\"1.0\" encoding=\"UTF-8\" standalone=\"yes\"?>"
-    "<problems xmlns=\"http://www.redhat.com/gss/strata\">"
-      "<link uri=\"http://access.redhat.com/\" rel=\"help\">The main Red Hat Support web site</link>"
-      "<property name=\"content\">an ABRT report</property>"
-      "<problem>"
-        "<property name=\"source\">a backtrace in the ABRT report</property>"
-        "<link uri=\"https://avalon-ci.gss.redhat.com/kb/docs/DOC-22029\" rel=\"suggestion\">[RHEL 5.3] EVO autocompletion lookup hang</link>"
-      "</problem>"
-    "</problems>"
-    );
+        log("ERR:%d", result->error);
+        log("MSG:'%s'", result->msg);
+        log("BODY:'%s'", result->body);
+        result->error = 0;
+        result->body = xstrdup(
+        "<?xml version=\"1.0\" encoding=\"UTF-8\" standalone=\"yes\"?>"
+        "<problems xmlns=\"http://www.redhat.com/gss/strata\">"
+          "<link uri=\"http://access.redhat.com/\" rel=\"help\">The main Red Hat Support web site</link>"
+          "<property name=\"content\">an ABRT report</property>"
+          "<problem>"
+            "<property name=\"source\">a backtrace in the ABRT report</property>"
+            "<link uri=\"https://avalon-ci.gss.redhat.com/kb/docs/DOC-22029\" rel=\"suggestion\">[RHEL 5.3] EVO autocompletion lookup hang</link>"
+          "</problem>"
+        "</problems>"
+        );
 #endif
-    if (result->error == 0 && result->body)
-    {
-        /* The message might contain URLs to known solutions and such */
-        char *hint = parse_response_from_RHTS_hint_xml2txt(result->body);
-        if (hint)
+        if (result->error == 0 && result->body)
         {
-            hint = append_to_malloced_string(hint, " ");
-            hint = append_to_malloced_string(hint,
-                    _("Do you still want to create a RHTSupport ticket?")
-            );
-            int create_ticket = ask_yes_no(hint);
-            free(hint);
-            if (!create_ticket)
-                goto ret;
+            /* The message might contain URLs to known solutions and such */
+            char *hint = parse_response_from_RHTS_hint_xml2txt(result->body);
+            if (hint)
+            {
+                hint = append_to_malloced_string(hint, " ");
+                hint = append_to_malloced_string(hint,
+                        _("Do you still want to create a RHTSupport ticket?")
+                );
+                int create_ticket = ask_yes_no(hint);
+                free(hint);
+                if (!create_ticket)
+                    goto ret;
+            }
         }
+        free_rhts_result(result);
+        result = NULL;
     }
-    free_rhts_result(result);
-    result = NULL;
 
     if (!(opts & OPT_t))
     {
-- 
1.8.2.1

